src/transformers/models/fastspeech2_conformer/modeling_fastspeech2_conformer.py

```diff
@@ -0,0 +1,1781 @@
+            normalize_before=config.decoder_normalize_before,
+            concat_after=config.decoder_concat_after,
+            positionwise_conv_kernel_size=config.positionwise_conv_kernel_size,
+            macaron_style=config.use_macaron_style_in_conformer,
+            use_cnn_module=config.use_cnn_in_conformer,
+            cnn_module_kernel=config.decoder_kernel_size,
+            utt_embed=config.utt_embed_dim,
+        )
+
+        self.speech_decoder_postnet = FastSpeech2ConformerSpeechDecoderPostnet(config)
+
+        self.criterion = FastSpeech2ConformerLoss(config)
+
+        self.post_init()
+
+    @replace_return_docstrings(output_type=FastSpeech2ConformerModelOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        input_ids: torch.LongTensor,
+        attention_mask: Optional[torch.LongTensor] = None,
+        spectrogram_labels: Optional[torch.FloatTensor] = None,
+        duration_labels: Optional[torch.LongTensor] = None,
+        pitch_labels: Optional[torch.FloatTensor] = None,
+        energy_labels: Optional[torch.FloatTensor] = None,
+        utterance_embedding: Optional[torch.FloatTensor] = None,
+        alpha: Optional[float] = 1.0,
+        lang_id: Optional[torch.LongTensor] = None,
+        return_dict: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+    ) -> Union[Tuple, FastSpeech2ConformerModelOutput]:
+        """
+        Args:
+            input_ids (`torch.LongTensor` of shape `(batch_size, sequence_length)`):
+                Input sequence of text vectors.
+            attention_mask (`torch.LongTensor` of shape `(batch_size, sequence_length)`, *optional*):
+                Mask to avoid performing convolution and attention on padding token indices. Mask values selected in
+                `[0, 1]`: 0 for tokens that are **masked**, 1 for tokens that are **not masked**.
+            spectrogram_labels (`torch.FloatTensor` of shape `(batch_size, max_spectrogram_length, num_mel_bins)`, *optional*):
+                Batch of padded target features.
+            duration_labels (`torch.LongTensor` of shape `(batch_size, sequence_length + 1)`, *optional*):
+                Batch of padded durations.
+            pitch_labels (`torch.FloatTensor` of shape `(batch_size, sequence_length + 1, 1)`, *optional*):
+                Batch of padded token-averaged pitch.
+            energy_labels (`torch.FloatTensor` of shape `(batch_size, sequence_length + 1, 1)`, *optional*):
+                Batch of padded token-averaged energy.
+            utterance_embedding (`torch.FloatTensor` of shape `(batch_size, embedding_dim)`, *optional*):
+                Tensor containing the utterance embeddings.
+            alpha (`float`, *optional*):
+                Alpha to control the speed of the spectrogram generation. Default is `1.0`.
+            lang_id (`torch.LongTensor`, *optional*):
```

Comment:
Renamed to lang_ids since the name is misleading, it actually has to be passed in the forward because it is different for different batches of audio. Below is the configuration var relevant to it
```py
num_languages (`int`, *optional*, defaults to `None`):
            Number of languages. If set to > 1, assume that the language ids will be provided as the input and use the
            languge id embedding layer.
```